### ISSUE TITLE
Is it possible to implement a method hook via entry_points?

### ISSUE THREAD

**Issue by bram2000** [pytest-dev/pluggy#285]
It seems from the example code that you can register a class with the plugin manager, and any hooks implemented as methods on that class will be registered. However, I can't seem to get this to work when using a separate python package, and using entry_points in setup.py. If I point the entry_point at the module where I have defined my plugin class, the register method does not attempt to look into any classes in the plugin, it only looks at the top level for matching function names.

Basically, can an implementation from a plugin loaded via setup.py entry_point, use the `self` reference? All of the doc examples where class methods are plugged, do so from within a single python package.

**Comment by bluetech** [comment]
You can register a class *instance*, not a class on its own (unless it just acts as a namespace with static/class methods perhaps). Entry points can point to an object, but it won't instantiate a class itself.

If you provide example code for your plugin and the entry point definition and what you want to happen, maybe we can understand better.

**Comment by RonnyPfannschmidt** [comment]
i think we can close this

pytest has prior art of this - when needing to register a configured/configuration free class instance, having a module level hook that instantiates the class is the way to go

a different valid solution is to implement a own entry-point loader

**Comment by bram2000** [comment]
Sorry for never commenting back on this issue. I did manage to get things working as I wanted, by instantiating the class at the module level, and pointing the entry_point at that instance. Thanks @bluetech and @RonnyPfannschmidt .

Happy to close.
